hebiros/include/hebiros/hebiros.cpp

```diff
@@ -208,8 +208,290 @@ bool Hebiros_Node::srv_add_group_from_urdf(
   return Hebiros_Node::srv_add_group_from_names(names_req, names_res);
 }
 
-//Subscriber callback which receives a joint state and sends that as a command to a group
-void Hebiros_Node::sub_command(const boost::shared_ptr<sensor_msgs::JointState const> data,
+//Add joint state values to a group command
+void Hebiros_Node::add_joint_command(GroupCommand* group_command,
+  sensor_msgs::JointState data, std::string group_name) {
+
+  std::shared_ptr<Group> group = groups[group_name];
+
+  for (int i = 0; i < data.name.size(); i++) {
+    if (group_joints[group_name].find(data.name[i]) != group_joints[group_name].end()) {
+
+      int joint_index = group_joints[group_name][data.name[i]];
+
+      if (i < data.position.size()) {
+        (*group_command)[joint_index].actuator().position().set(data.position[i]);
+      }
+      if (i < data.velocity.size()) {
+        (*group_command)[joint_index].actuator().velocity().set(data.velocity[i]);
+      }
+      if (i < data.effort.size()) {
+        (*group_command)[joint_index].actuator().effort().set(data.effort[i]);
+      }
+    }
+    else {
+      ROS_WARN("Unable to find joint: %s.  Command will not be sent.", data.name[i].c_str());
+    }
+  }
+}
+
+//Add settings values to a group command
+void Hebiros_Node::add_settings_command(GroupCommand* group_command,
+  SettingsMsg data, std::string group_name) {
+
+  std::shared_ptr<Group> group = groups[group_name];
+
+  for (int i = 0; i < data.name.size(); i++) {
+    if (group_joints[group_name].find(data.name[i]) != group_joints[group_name].end()) {
```

Comment:
The basic pattern here (iterate and search for a module by name) is used quite a bit in this file; this makes me wonder if there is a good way to refactor this.